backend/wireguard/wireguard.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2019 flannel authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package wireguard
+
+import (
+	"fmt"
+	"time"
+
+	"encoding/json"
+	"sync"
+
+	"github.com/flannel-io/flannel/backend"
+	"github.com/flannel-io/flannel/pkg/ip"
+	"github.com/flannel-io/flannel/subnet"
+	"golang.org/x/net/context"
+)
+
+func init() {
+	backend.Register("wireguard", New)
+}
+
+type WireguardBackend struct {
+	sm       subnet.Manager
+	extIface *backend.ExternalInterface
+	networks map[string]*network
```

Comment:
We use WireGuard to configure a device which routes a complete network. But you could also configure WireGuard to share one subnet between several nodes and have just a single IP per node.

Reply:
What I wonder is: what is the use case in which the map `WireguardBackend.networks` will have more than 1 item? IIUC, if we deploy default flannel, that map will only have 1 item regardless of the number of nodes, right?

Reply:
My first reply to this is just completely misleading. I removed this field as I implemented the dual-stack support because I noticed the field wasn't used in wireguard backend. TBH I don't know why I added this field two years ago ;)